### ISSUE TITLE
Incompatibility with ZED-F9R HPS 1.30

### ISSUE THREAD

**Issue by PaulZC** [sparkfun/OpenLog_Artemis_GNSS_Logger#28]
As reported in the forum:

https://forum.sparkfun.com/viewtopic.php?f=172&t=58852

**Comment by PaulZC** [comment]
So far, I'm not able to replicate this issue. With ZED-F9R (HPS v1.30) connected to an OLA (v2.4, without IMU) I see:

```
Artemis OpenLog v2.4
Finding the next available log file.
This could take a long time if the SD card contains many existing log files.
Logging to: dataLog00003.TXT
Finding the next available log file.
This could take a long time if the SD card contains many existing log files.
Logging to: serialLog00000.TXT
ICM-20948 failed to init.
SD card online
Data logging online
Serial logging online
IMU offline - or not present
Identifying Qwiic Muxes...
Identifying Qwiic Devices...
Autodetect complete
GPS-ublox online at address 0x42
Content-Type: text/csv

rtcDate,rtcTime,gps_Date,gps_Time,gps_Lat,gps_Long,gps_Alt,gps_SIV,gps_FixType,gps_GroundSpeed,gps_Heading,gps_pDOP,output_Hz,
01/01/2000,00:04:07.32,09/11/2022,00:00:02.000,0,0,0,0,0,0,0,9999,0.971,
01/01/2000,00:04:08.37,09/11/2022,00:00:03.000,0,0,0,0,0,0,0,9999,0.985,
01/01/2000,00:04:09.36,09/11/2022,00:00:04.000,0,0,0,0,0,0,0,9999,0.993,
01/01/2000,00:04:10.36,09/11/2022,00:00:05.000,0,0,0,0,0,0,0,9999,0.990,
01/01/2000,00:04:11.38,09/11/2022,00:00:06.000,0,0,0,0,0,0,0,9999,0.992,
01/01/2000,00:04:12.37,09/11/2022,00:00:07.000,0,0,0,0,0,0,0,9999,0.995,
01/01/2000,00:04:13.37,09/11/2022,00:00:08.000,0,0,0,0,0,0,0,9999,0.996,
```

**Comment by PaulZC** [comment]
With an older OLA - with IMU:

```
Artemis OpenLog v2.4
Logging to: dataLog00006.TXT
Finding the next available log file.
This could take a long time if the SD card contains many existing log files.
Logging to: serialLog00000.TXT
SD card online
Data logging online
Serial logging online
IMU online
Identifying Qwiic Muxes...
Identifying Qwiic Devices...
Autodetect complete
GPS-ublox online at address 0x42
Content-Type: text/csv

rtcDate,rtcTime,aX,aY,aZ,gX,gY,gZ,mX,mY,mZ,imu_degC,gps_Date,gps_Time,gps_Lat,gps_Long,gps_Alt,gps_SIV,gps_FixType,gps_GroundSpeed,gps_Heading,gps_pDOP,output_Hz,
01/01/2000,00:00:51.17,212.89,-75.20,-985.84,-1.93,-0.27,-2.81,-58.65,3.60,-81.75,27.84,09/11/2022,00:00:02.000,0,0,0,0,0,0,0,9999,0.962,
01/01/2000,00:00:52.22,195.80,-83.50,-980.96,-1.52,2.15,-1.04,-60.15,3.30,-79.50,27.89,09/11/2022,00:00:03.000,0,0,0,0,0,0,0,9999,0.980,
01/01/2000,00:00:53.22,193.85,-82.03,-994.14,1.07,-0.28,0.33,-59.85,3.45,-78.90,27.69,09/11/2022,00:00:04.000,0,0,0,0,0,0,0,9999,0.990,
01/01/2000,00:00:54.22,199.71,-78.61,-978.03,-0.59,-3.79,-1.66,-57.45,3.30,-80.55,28.03,09/11/2022,00:00:05.000,0,0,0,0,0,0,0,9999,0.985,
01/01/2000,00:00:55.24,203.13,-89.84,-984.86,1.11,-1.92,-0.37,-57.45,4.50,-82.50,28.17,09/11/2022,00:00:06.000,0,0,0,0,0,0,0,9999,0.990,
```

**Comment by baegas20** [comment]
Artemis OpenLog GNSS v2.2
Created log file: dataLog00041.ubx
SD card online
Data logging online
GNSS offline



Returning to logging...
detectQwiicDevices: device found at address 0x42
createFileBuffer: Warning. fileBufferSize is zero. Data logging is not possible.

Sending: CLS:CFG ID:RATE Len: 0x0 Payload:
sendCommand: Waiting for ACK response
checkUbloxI2C: Reading 10 bytes
processUBX: NACK received: Requested Class: 0x6 Requested ID: 0x8
Incoming: Size: 2 Received: CLS:ACK ID:0x0 Len: 0x2 Payload: 6 8
packetCfg now valid
packetAck now valid
waitForACKResponse: data was NOTACKNOWLEDGED (NACK) after 13 msec

Sending: CLS:CFG ID:RATE Len: 0x0 Payload:
sendCommand: Waiting for ACK response
checkUbloxI2C: Reading 10 bytes
processUBX: NACK received: Requested Class: 0x6 Requested ID: 0x8
Incoming: Size: 2 Received: CLS:ACK ID:0x0 Len: 0x2 Payload: 6 8
packetCfg now valid
packetAck now valid
waitForACKResponse: data was NOTACKNOWLEDGED (NACK) after 13 msec

Sending: CLS:CFG ID:RATE Len: 0x0 Payload:
sendCommand: Waiting for ACK response
checkUbloxI2C: Reading 10 bytes
processUBX: NACK received: Requested Class: 0x6 Requested ID: 0x8
Incoming: Size: 2 Received: CLS:ACK ID:0x0 Len: 0x2 Payload: 6 8
packetCfg now valid
packetAck now valid
waitForACKResponse: data was NOTACKNOWLEDGED (NACK) after 13 msec
beginSensors: no qwiic devices detected!

**Comment by PaulZC** [comment]
Hi @baegas20 / Dong-Geulami,

Thank you for the update. You did it right - thank you.

OK. I understand the problem. With HPS 1.30 the UBX-CFG-RATE message is deprecated (no longer supported). It is being Not-ACKnowledged (NACK).

But, I am also confused - because the 1.30 module I have here works (it _does_ respond to and acknowledge CFG-RATE).

Please take a photo of the top of your ZED-F9R and post it here. I would like to see what hardware version it is (e.g. -01B-00).

Thank you,
Paul

**Comment by baegas20** [comment]
Hello Paul
Thanks for the reply, I have attached the picture you said
Thank you


![20230107_091606](https://user-images.githubusercontent.com/63548387/211121458-b83e65d4-9569-4a28-9ad1-9d9591df9217.jpg)

**Comment by PaulZC** [comment]
Hello @baegas20 / Dong-Geulami,

Ah! OK! You are using the OLA GNSS Logger firmware. Sorry! I thought you were using the standard OLA firmware.

OK. I understand completely now. Yes, the GNSS Logger firmware will not work with HPS 1.30 - as it uses some messages which are now deprecated (no longer supported). I will need to change the firmware so it uses v3 of our u-blox GNSS library:

https://github.com/sparkfun/SparkFun_u-blox_GNSS_v3

v3 of the library is public - but it is still "work in progress" and is not yet released. I am still adding and testing examples. As soon as I have added the missing examples, I will update the OLA GNSS Logger. It will take me a few days. Thank you for your patience and support.

Best wishes,
Paul

**Comment by baegas20** [comment]
Hello Paul
Thanks for checking, looking forward to the firmware update and hope it works well
thank you

**Comment by PaulZC** [comment]
Hello @baegas20 / Dong-Geulami,

Please try this BETA version of the firmware:

https://github.com/sparkfun/OpenLog_Artemis_GNSS_Logger/blob/main/Binaries/OpenLog_Artemis_GNSS_Logging-V10-v30_BETA.bin

The upgrade instructions are here:

https://github.com/sparkfun/OpenLog_Artemis_GNSS_Logger/blob/main/UPGRADE.md

Please let me know if it works OK for you.

Thank you,
Paul

**Comment by baegas20** [comment]
Hello @PaulZC
Firmware V30 beta works fine
Raw data recording is also good. There are no data errors or interruptions so far. If there is a problem while using it, I will leave a message here.
Thank you sincerely

**Comment by PaulZC** [comment]
Resolved in v3.0: https://github.com/sparkfun/OpenLog_Artemis_GNSS_Logger/releases/tag/v3.0

Closing...
